ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcService.java

```diff
@@ -118,23 +144,76 @@ private GrpcService(RaftServer raftServer, Supplier<RaftPeerId> idSupplier, int
 
     this.clientProtocolService = new GrpcClientProtocolService(idSupplier, raftServer);
 
+    final int port = serverConfig.getPort();
     this.serverInterceptor = new MetricServerInterceptor(
         idSupplier,
         JavaUtils.getClassSimpleName(getClass()) + "_" + port
     );
 
-    NettyServerBuilder nettyServerBuilder = NettyServerBuilder.forPort(port)
+    final boolean separateAdminServer = !serverConfig.equals(adminServerConfig);
+    final boolean separateClientServer = !serverConfig.equals(clientServerConfig);
```

Comment:
Let's compare getPort(), i.e.
```
    final boolean separateAdminServer = adminServerConfig.getPort() != port;
    final boolean separateClientServer = clientServerConfig.getPort() != port;
```

Reply:
Should I get rid of `ServerConfig` class completely?  Doesn't seem to add much value after removing `equals()`: it only bundles 2 parameters for the `private GrpcService` constructor.

Reply:
I am fine for removing ServerConfig if it is not useful.